### PR TITLE
[docs] Improve docs and address some ahrefs reported 404s 

### PR DESCRIPTION
**Pull request by Janpot**
Address some ahrefs issues

* Fix 404s
* Add missing title  and description

## TO DO
- [ ] New og image
- [ ] better copy for the titles/descriptions?

### CODE REVIEW

File: docs/data/toolpad/core/introduction/overview.md [hunk 1]
```diff
@@ -4,7 +4,7 @@ title: Toolpad Overview
 
 # Toolpad Core
 
-<p class="description">A framework to build full-stack apps, fast. It can be used as a Next.js starter, or integrated within an existing React project to build dashboards and internal tools. </p>
```
Comment (Janpot) [inline docs/data/toolpad/core/introduction/overview.md:7]: @prakhargupta1 This title is reported as too long. We need a shorter, more descriptive one.

File: docs/pages/toolpad/index.js [hunk 2]
```diff
@@ -14,6 +15,11 @@ import StudioIntro from '../../src/components/landing/StudioIntro';
 export default function Home() {
   return (
     <BrandingCssVarsProvider>
+      <Head
+        title="Toolpad Core: Open-source dashboard framework for React"
+        description="Toolpad Core offers the components needed for your next admin panel and internal tools project."
```
Comment (Janpot) [inline docs/pages/toolpad/index.js:20]: @prakhargupta1 This title/description was missing, might need better copy. Also may want to update the twitter card image

File: docs/data/toolpad/core/introduction/overview.md (hunk 1 above)
Comment (prakhargupta1) [reply docs/data/toolpad/core/introduction/overview.md:7]: Changed it to: A set of full-stack components to build react dashboards.

**Comment by prakhargupta1** [comment]
There is a 404 warning for https://mui.com/home coming from https://mui.com/toolpad/core/react-page-container/. Olivier's suggestion: For /home we can replace this with /inbox we already have a case for those in the docs.

File: docs/data/toolpad/core/components/use-notifications/use-notifications.md [hunk 3]
```diff
@@ -4,9 +4,9 @@ title: useNotifications
 components: NotificationsProvider
 ---
 
-# Notifications
+# useNotifications
```
Comment (Janpot) [inline docs/data/toolpad/core/components/use-notifications/use-notifications.md:7]: I think we can keep "Notifications" and instead change the `title` property above

File: packages/toolpad-core/README.md [hunk 4]
```diff
@@ -5,7 +5,7 @@
 
 <h1 align="center">Toolpad Core</h1>
 
-Toolpad Core is a React framework for building admin dashboards, internal tools, and B2B web applications. It is built on top of Next.js and offers a suite of components such as layout, navigation, authentication, and data management interfaces to help you build scalable dashboards fast.
+Toolpad Core is a set of full-stack components for building react dashboards, internal tools, and B2B web applications. It is built on top of Next.js and offers a suite of components such as layout, navigation, authentication, and data management interfaces to help you build scalable dashboards fast.
```
Comment (Janpot) [inline packages/toolpad-core/README.md:8]:
```suggestion
Toolpad Core is a set of full-stack components for building dashboards, internal tools, and B2B web applications with React. It is built on top of Next.js and offers a suite of components such as layout, navigation, authentication, and data management interfaces to help you build scalable dashboards fast.
```

File: docs/data/toolpad/core/pages.ts [hunk 5]
```diff
@@ -85,7 +85,7 @@ const pages: MuiPage[] = [
           },
           {
             pathname: '/toolpad/core/react-persistent-state',
-            title: 'Peristent State',
+            title: 'Peristent storage state',
```
Comment (Janpot) [inline docs/data/toolpad/core/pages.ts:88]: 1. typo in "peristent" => I make that "Persisted"
1. we're planning to put url query state under this umbrella as well, so I'd leave out "storage"

```suggestion
            title: 'Persisted state',
```

File: docs/data/toolpad/core/introduction/overview.md [hunk 6]
```diff
@@ -4,25 +4,21 @@ title: Toolpad Overview
 
 # Toolpad Core
 
-<p class="description">A framework to build full-stack apps. You can use it as a Next.js starter, or integrate it within an existing React project to build dashboards and internal tools.</p>
+<p class="description">A set of full-stack components to build react dashboards.</p>
```
Comment (Janpot) [inline docs/data/toolpad/core/introduction/overview.md:7]:
```suggestion
<p class="description">A set of full-stack components to build dashboards with React.</p>
```

**Review by bharatkashyap — APPROVED** [review]

**Pull request merged**